hostlabels/syncer.go

```diff
@@ -95,8 +127,11 @@ func sync(kClient *kubernetesclient.Client, metadataClient metadata.Client, c *c
 		for changed {
 			node, err := getKubeNode(kClient, host.Hostname)
 			if err != nil {
-				log.Errorf("Error getting node: [%s] by name from kubernetes: [%v]", host.Hostname, err)
-				continue
+				node, ok = rancherHostNodeMap[host.Hostname]
```

Comment:
the reason behind fetching the node for the second time is - to get the most recent node object of k8s. So rethinking if our map should really be key=`hostName`, value=`nodeName`. So you can fetch node by name to get more "fresh" copy